src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedLogicalPlanPrinterImpl2.java

```diff
@@ -181,19 +105,19 @@ else if ( rootOp instanceof LogicalOpUnion ) {
 	protected void printOperatorInfoForBGPAdd ( final LogicalOpBGPAdd op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
 		out.append( indentLevelString + "bgpAdd (" + op.getID() + ") " );
 		out.append( System.lineSeparator() );
-		printFederationMember( op.getFederationMember(), indentLevelStringForOpDetail + levelIndentBase, out );
-		printSPARQLGraphPattern( op.getBGP(), indentLevelStringForOpDetail + levelIndentBase, out );
-		out.append( indentLevelStringForOpDetail + levelIndentBase );
-		out.append( System.lineSeparator() );
+		printFederationMember( op.getFederationMember(), indentLevelStringForOpDetail + singleBase, out );
+		printSPARQLGraphPattern( op.getBGP(), indentLevelStringForOpDetail + singleBase, out );
+		//out.append( indentLevelStringForOpDetail + singleBase );
+		//out.append( System.lineSeparator() );
```

Comment:
Not needed anymore? Then, remove.

However, I guess that these two code lines are indeed needed after the fixing the issue that have pointed out in my comment above about the `printSPARQLGraphPattern` function.